tests/drivers/gpio/gpio_ite_it8xxx2_v2/src/main.c

```diff
@@ -246,30 +251,54 @@ ZTEST(gpio_ite_it8xxx2_v2, test_interrupt_level_active)
 {
 	zassert_true(device_is_ready(gpio_dev));
 	zassert_ok(gpio_pin_configure(gpio_dev, TEST_PIN, GPIO_INPUT | GPIO_ACTIVE_HIGH));
-	zassert_equal(gpio_pin_interrupt_configure(gpio_dev, TEST_PIN, GPIO_INT_LEVEL_ACTIVE),
-		      -ENOTSUP);
+
+	gpio_init_callback(&callback_struct, &callback, BIT(TEST_PIN));
+	zassert_ok(gpio_add_callback(gpio_dev, &callback_struct));
+	zassert_ok(gpio_pin_interrupt_configure(gpio_dev, TEST_PIN, GPIO_INT_LEVEL_ACTIVE));
 	zexpect_equal(reg_gpotr, 0, "reg_gpotr=%x", reg_gpotr);
 	zexpect_equal(reg_p18scr, 0);
 	zexpect_equal(reg_gpcr[TEST_PIN], GPCR_PORT_PIN_MODE_INPUT, "reg_gpcr[%d]=%x", TEST_PIN,
 		      reg_gpcr[TEST_PIN]);
 	zexpect_equal(reg_wubemr, 0, "reg_wubemr=%x", reg_wubemr);
 	zexpect_equal(reg_wuemr, 0, "reg_wuemr=%x", reg_wuemr);
-	zexpect_equal(reg_wuesr, 0, "reg_wuesr=%x", reg_wuesr);
+	zexpect_equal(reg_wuesr, TEST_MASK, "reg_wuesr=%x", reg_wuesr);
+	reg_wuesr = 0;
+	k_sleep(K_MSEC(100));
+	zexpect_equal(callback_called, 0, "callback_called=%d", callback_called);
+
+	reg_gpdmr = BIT(TEST_PIN);
+	/* Mock the hardware interrupt. */
+	posix_sw_set_pending_IRQ(TEST_IRQ);
+	k_sleep(K_MSEC(100));
+	zexpect_equal(callback_called, 5, "callback_called=%d", callback_called);
 }
 
 ZTEST(gpio_ite_it8xxx2_v2, test_interrupt_level_inactive)
 {
 	zassert_true(device_is_ready(gpio_dev));
 	zassert_ok(gpio_pin_configure(gpio_dev, TEST_PIN, GPIO_INPUT | GPIO_ACTIVE_HIGH));
-	zassert_equal(gpio_pin_interrupt_configure(gpio_dev, TEST_PIN, GPIO_INT_LEVEL_INACTIVE),
-		      -ENOTSUP);
+
+	gpio_init_callback(&callback_struct, &callback, BIT(TEST_PIN));
+	zassert_ok(gpio_add_callback(gpio_dev, &callback_struct));
+	zassert_ok(gpio_pin_interrupt_configure(gpio_dev, TEST_PIN, GPIO_INT_LEVEL_INACTIVE));
 	zexpect_equal(reg_gpotr, 0, "reg_gpotr=%x", reg_gpotr);
 	zexpect_equal(reg_p18scr, 0);
 	zexpect_equal(reg_gpcr[TEST_PIN], GPCR_PORT_PIN_MODE_INPUT, "reg_gpcr[%d]=%x", TEST_PIN,
 		      reg_gpcr[TEST_PIN]);
 	zexpect_equal(reg_wubemr, 0, "reg_wubemr=%x", reg_wubemr);
-	zexpect_equal(reg_wuemr, 0, "reg_wuemr=%x", reg_wuemr);
-	zexpect_equal(reg_wuesr, 0, "reg_wuesr=%x", reg_wuesr);
+	zexpect_equal(reg_wuemr, TEST_MASK, "reg_wuemr=%x", reg_wuemr);
+	zexpect_equal(reg_wuesr, TEST_MASK, "reg_wuesr=%x", reg_wuesr);
+	reg_wuesr = 0;
+	k_sleep(K_MSEC(100));
+	/* The interrupt was already active when we started. */
+	zexpect_equal(callback_called, 5, "callback_called=%d", callback_called);
```

Comment:
I think a better test would be to set the `reg_gpdmr = BIT(TEST_PIN)` prior to `gpio_pin_interrupt_configure`.  This simulates the typical hardware interaction where the interrupt line will be inactive prior to enabling the interrupt.

Reply:
I did that in  test_interrupt_level_active. I just wanted to handle both of those scenarios.  If you think it necessary, I could make this 4 tests.

1) level active, not firing before configure
2) level inactive, not firing before configure
3) level active, firing before configure
4) level inactive, firing before configure

Right now I have case 1 and 4.

Reply:
Test cases 1 and 4 seem sufficient.  Can you add a comment documenting this?

Reply:
done